rewards/tx_utils.py

```diff
@@ -116,6 +116,29 @@ def get_priority_fee(
     return priority_fee
 
 
+def check_tx_receipt(web3: Web3, tx_hash: HexBytes, timeout: int, tries: int = 5):
+    tx_found = False
+    attempt = 0
+    error = None
+    while not tx_found and attempt < tries:
+        try:
+            web3.eth.wait_for_transaction_receipt(tx_hash, timeout=timeout)
+            msg = f"Transaction {tx_hash} succeeded!"
+            send_message_to_discord("Transaction Success", msg, [], "Rewards Bot")
+        except Exception as e:
+            msg = f"Error waiting for {tx_hash}. Error: {e}. \n Retrying..."
+            attempt += 1
+            error = e
+            logger.error(msg)
+            send_message_to_discord("Transaction Error", msg, [], "Rewards Bot")
```

Comment:
I'd only send a message to discord on total failure, we can keep the logs for debugging but this will just make noise IMO